rusk-profile/src/lib.rs

```diff
@@ -209,9 +209,20 @@ pub fn verify_common_reference_string(buff: &[u8]) -> bool {
 
 pub fn keys_for(crate_name: &str) -> Keys {
     use cargo_lock::{Lockfile, Package};
-    // FIXME: This will only work for workspaces.
-    let lockfile =
-        Lockfile::load("./../Cargo.lock").expect("Cargo.lock should exists");
+
+    let mut lockfile = env::current_dir().expect("Failed to get current dir");
+    loop {
+        lockfile.push("Cargo.lock");
+        if lockfile.as_path().exists() {
+            break;
+        }
+
+        if !lockfile.pop() || !lockfile.pop() {
+            panic!("Failed to fetch Cargo.lock from upper dir!");
+        }
+    }
```

Comment:
this is a bit hacky. I think you can have a better code if, instead having the starting path that you mutate every time and check with double `pop`, you have two separate path. The `folder` you're currently inspect, and the path containing the folder + the `Cargo.lock` that you want to check.

So, first I would create a local private function to get the `Cargo.lock` (something like `fn get_lockfile_path() -> PathBuf` or similar).

Then, instead of `push` inside the loop I would use `join`, that does not alter the current path with the `Cargo.lock` but returns a new one. You then check if the new path exists, and if it does you return it. If not, you can single `pop` the current (since it doesn't contain the `"Cargo.toml"` part), and if it's false means you already reached the root without finding a `Cargo.toml` and you can panic.